src/graphql.tsx

```diff
@@ -635,7 +635,10 @@ class ObservableQueryRecycler {
   public recycle (observableQuery: ObservableQuery<any>): void {
     // Stop the query from polling when we recycle. Polling may resume when we
     // reuse it and call `setOptions`.
-    observableQuery.stopPolling();
+    observableQuery.setOptions({
+      fetchPolicy: 'cache-only',
```

Comment:
I think the solution is to update line 672 so that it sets the default fetch policy if no other fetch policy was found. So:

```ts
observableQuery.setOptions({
  ...options,
  fetchPolicy: options.fetchPolicy || 'xyz',
});
```

Where `'xyz'` is the default `fetchPolicy`. This should guarantee that the `cache-only` policy is reset.